src/data_info/services/entry.clj

```diff
@@ -239,22 +235,25 @@
         offset      (if-not (nil? offset) offset 0)]
     (http-response/ok
       (paged-dir-listing
-         cm user path entity-type badies sort-field sort-dir offset limit info-type))))
+         irods user path entity-type badies sort-field sort-dir offset limit info-type))))
 
 
 (defn- get-path-attrs
-  [cm zone path-in-zone user]
-  (let [path (file/rm-last-slash (irods/abs-path zone path-in-zone))]
+  [irods zone path-in-zone user]
+  (let [path (file/rm-last-slash (irods/abs-path zone path-in-zone))
+        object-type (rods/object-type irods user zone path)]
     (jv/validate-path-lengths path)
-    (duv/path-exists cm path)
-    {:path           path
-     :is-dir?        (item/is-dir? cm path)}))
+    (if (= @object-type :none)
+      (throw+ {:error_code error/ERR_DOES_NOT_EXIST
+               :path path})
```

Comment:
This is basically what jargon's `exists` did already -- grab the object type, if it throws an exception or returns something we can't identify, say it doesn't exist -- the exception handling is in the other PRs, they just turn it into this `:none`.